daemon/cluster/cluster.go

```diff
@@ -334,8 +334,9 @@ func (c *Cluster) Cleanup() {
 		c.mu.Unlock()
 		return
 	}
-	defer c.mu.Unlock()
 	state := c.currentNodeState()
+	c.mu.Unlock()
```

Comment:
I saw the comment about "currentNodeState should not be called without a read lock", but I do not understand it.
I believe that was added just because `c.currentNodeState()` will call `c.nr.State()`, so to make sure the `c.nr` won't change in between.
If possible we should avoid to hold locks around function executions.

I think we should change the code here to
```
c.mu.Lock()
node := c.nr
c.mu.Unlock()
...
state := nr.State()
```